libs/commons/Immediate_bitfield.ml

```diff
@@ -0,0 +1,52 @@
+(* Andre Kuhlenschmidt
+ *
+ * Copyright (C) 2023 r2c
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+
+(* This library implements a bitfield that can be stored as an
```

Comment:
better to move some useful info for the user of the API in the .mli rather than the .ml
In the .ml we can add comments directed to the implementor or maintainer of the API.